[Memory Snapshot][Viz] Add Allocator Settings Tab

aaronenyeshi opened the pull request:
Summary: Since we are storing the allocator settings in the snapshot files for awhile now (since https://github.com/pytorch/pytorch/pull/119404), we can expose this to users with a new tab in the visualizer.

Test Plan:
Ran it locally:
![image](https://github.com/user-attachments/assets/5f79ccd0-fe1c-4e42-bb58-106d8f3cccd6)

Differential Revision: D60673548

Pulled By: aaronenyeshi

zdevito commented on torch/utils/viz/MemoryViz.js:
```diff
@@ -1271,6 +1271,36 @@ function create_trace_view(
   plot.set_delegate(delegate);
 }
 
+function create_settings_view(dst, snapshot, device) {
```
To make this easy to maintain maybe just do:
`settings_div.append('pre').text(JSON.stringify(snapshot.allocator_settings))`. It will handle whatever nested stuff we put in there without having to update this page.

aaronenyeshi replied:
@zdevito I tried this out, but it doesn't not look very good. Everything is put into one line.
![image](https://github.com/user-attachments/assets/9dbdd2f9-23aa-4a02-98a0-011350bb4a51)

aaronenyeshi replied:
I've discovered JSON pretty printing, and it looks so much better, let's go with that one:
![image](https://github.com/user-attachments/assets/8b3df71c-f987-4604-b4b9-7b2eb7d4a75c)

aaronenyeshi commented:
@pytorchbot rebase

aaronenyeshi commented:
@pytorchbot merge

aaronenyeshi committed:
[Memory Snapshot][Viz] Add Allocator Settings Tab (#132518)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/132518

Since we are storing the allocator settings in the snapshot files for awhile now (since D53536199), we can expose this to users with a new tab in the visualizer.

Test Plan:
Ran it locally:

 {F1802124482}

Reviewed By: tianfengfrank

Differential Revision: D60673548

Pulled By: aaronenyeshi

aaronenyeshi commented:
@pytorchbot merge
